include/revng/AutoEnforcer/AutoEnforcerLibraryRegistry.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringMap.h"
+
+#include "revng/AutoEnforcer/PipelineLoader.h"
+
+namespace AutoEnforcer {
+
+class AutoEnforcerLibraryRegistry {
+public:
+  AutoEnforcerLibraryRegistry() { getInstances().push_back(this); }
```

Comment:
I'd opt for having a private constructor + factory method + static field for the registry.
But no biggie.

Reply:
i can't put private constructos since it must be extended. i put it protected instead.